aries_cloudagent/protocols/coordinate_mediation/v1_0/manager.py

```diff
@@ -74,11 +75,37 @@ async def _create_routing_did(self) -> DIDInfo:
         await storage.add_record(record)
         return info
 
-    async def receive_request(self, request: MediationRequest) -> MediationRecord:
-        """Create a new mediation record to track this request."""
+    async def receive_request(self,
+                              conn_id: str,
+                              request: MediationRequest
+                              ) -> MediationRecord:
+        """Create a new mediation record to track external request."""
+        role = MediationRecord.ROLE_SERVER
+        record = await self._store_request(request, conn_id, role)
+        return record
+
+    async def prepare_request(self,
+                              conn_id: str,
+                              request: MediationRequest
+                              ) -> MediationRecord:
+        """Create a new mediation record to track internal request."""
+        role = MediationRecord.ROLE_CLIENT
+        record = await self._store_request(request, conn_id, role)
+        return record
+
+    async def _store_request(self,
+                             request: MediationRequest,
+                             _conn_id: str,
+                             _role: str
+                             ) -> MediationRecord:
+        """Create a new mediation record to track a request."""
         # TODO: Determine if terms are acceptable
+        connection_record = await ConnectionRecord.retrieve_by_id(
+            self.context, _conn_id
+        )
         record = MediationRecord(
-            connection_id=self.context.connection_record.connection_id,
+            role=_role,
+            connection_id=connection_record.connection_id,
```

Comment:
I think I prefer a little redundancy to having this `_store_request` method being split out.

Reply:
I split the logic out because it would be a direct copy-paste of the same code inside recieve_request and prepare_request. We could have redundancy or even push the logic into the routes.py handler. which do you prefer?

Reply:
The point I take issue with is that you've condensed just two method calls into one method call. It is several lines but in this case again, I prefer the redundancy and clarity.

Reply:
I understand. I will change this back.